lstchain/mc/mc.py

```diff
@@ -66,63 +67,89 @@ def int_diff_sp(emin, emax, sp_idx, e0):
 
     return integral_e
 
-def rate(emin, emax, sp_idx, cone, area, norm, e0):
+def rate(shape, emin, emax, param, cone, area):
     """
     Calculates the rate of events for a power-law distribution,
     in a given energy range, collection area and solid angle
 
     Parameters
     ----------
+    shape: `string` weighted spectrum shape
     emin:  `float`  minimum energy
     emax:  `float`  maximum energy
-    sp_idx:`float`  spectral index of the power-law distribution
+    param: `dict` with weighted spectral parameters
     cone:  `float`  angle [deg] for the solid angle cone
     area:  `float`  collection area [cm**2]
-    norm:  `float`  normalization of the differential energy spectrum at e0
-    e0:    `float`  normalization energy
+
+    if shape is 'PowerLaw':
+    param should include 'f0','e0','alpha'
+    dFdE  = f0 * np.power(E / e0, alpha)
+
+    if shape is 'LogParabola':
```

Comment:
Hi @rlopezcoto ,
For the first case(`LogParabola` without `beta`), we will get error.
But the second case(`PowerLaw` with `beta`), no error and just other parameters are used for the calcurlation.
Would it be better to add the function to check if the `param` match the `shape`?

Reply:
what kind of error in the first case? (a generic python one I guess).
It would be great if you could check that we are giving the correct parameters depending on the function introduced.

Reply:
> what kind of error in the first case? (a generic python one I guess).

Yes, just a KeyError.
```
>>> crab_par_hegra
{'f0': <Quantity 2.83e-11 1 / (cm2 s TeV)>, 'alpha': -2.62, 'e0': <Quantity 1. TeV>}
>>> rate_ = rate("LogParabola", 0.1 * u.TeV, 100 * u.TeV, crab_par_hegra, 0, 0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cta/Work/Install/anaconda3/envs/cta-dev/lib/python3.7/site-packages/lstchain-0.4.5.post9+git66dd663-py3.7.egg/lstchain/mc/mc.py", line 107, in rate
    log_parabola =  LogParabolaSpectralModel.from_log10(amplitude=param['f0'], reference=param['e0'], alpha=-1*param['alpha'], beta=-1*param['beta'])
KeyError: 'beta'
```

> It would be great if you could check that we are giving the correct parameters depending on the function introduced.

I'll introduce such a function in a few days.